node/full_test.go

```diff
@@ -6,29 +6,73 @@ import (
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+
+	"github.com/celestiaorg/celestia-core/abci/example/kvstore"
+	core_node "github.com/celestiaorg/celestia-core/node"
+	rpctest "github.com/celestiaorg/celestia-core/rpc/test"
+	"github.com/celestiaorg/celestia-node/node/p2p"
+	"github.com/celestiaorg/celestia-node/node/rpc"
 )
 
 func TestNewFull(t *testing.T) {
-	nd, err := NewFull(DefaultConfig())
+	coreNode := startCoreNode()
+	endpoint := coreNode.Config().RPC.ListenAddress
+	protocol, ip := endpoint[:3], endpoint[6:]
```

Comment:
nit: you can make code cleaner if you hide endpoint parsing in startCoreNode by making it return 3 vars.

Reply:
another nit: `startCoreNode` is used(copied) in multiple places already so can be moved to some test utilities folder.